python/whylogs/core/constraints/metric_constraints.py

```diff
@@ -282,17 +405,14 @@ def generate_constraints_report(
 
                 results.append(metric_report)
 
-        metric_names = self.dataset_constraints.keys()
```

Comment:
Line 267/390 is a bug. If `column_constraints` is empty but `dataset_constraints` is not (even in the old version where `dataset_constraint` contained `MetricConstraint`s), the reports for the dataset constraints would not be generated.